src/neuroconv/datainterfaces/ecephys/ced/ceddatainterface.py

```diff
@@ -9,12 +9,20 @@ def _test_sonpy_installation() -> None:
     get_package(package_name="sonpy", excluded_platforms_and_python_versions=dict(darwin=["3.7"]))
 
 
-class CEDRecordingInterface(BaseRecordingExtractorInterface):
-    """Primary data interface class for converting data from CED (Cambridge Electronic Design)."""
+class LazyExtractorClass(type(BaseRecordingExtractorInterface), type):
+    def __getattribute__(self, name):
+        if name == "RX":
+            _test_sonpy_installation()
+            spikeinterface = get_package(package_name="spikeinterface")
+            RX = spikeinterface.extractors.CedRecordingExtractor
+            self.RX = RX
```

Comment:
Couple touchy things to note - the attribute must also be 'set' at this call in order for it to persist later. Otherwise an instance of the class will not have it set (calling `self.RX` on an object does not itself seem to trigger `getattr(cls, "RX")`.